src/DynamoCore/Core/PulseMaker.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+
+using Dynamo.Core.Threading;
+using Dynamo.Models;
+
+namespace Dynamo.Core
+{
+    public class PulseMaker
+    {
+        #region Class Data Members and Properties
+
+        private readonly Timer internalTimer;
+        private readonly DynamoModel dynamoModel;
+
+        // PulseMaker internal state management objects.
+        private readonly object stateMutex;
+        private bool evaluationInProgress = false;
+        private bool evaluationRequestPending = false;
+
+        internal int TimerPeriod { get; private set; }
+
+        #endregion
+
+        #region Public Class Operational Methods
+
+        /// <summary>
+        /// An internal constructor to ensure PulseMaker object can only
+        /// be instantiated from within DynamoModel (i.e. DynamoCore.dll).
+        /// </summary>
+        /// <param name="dynamoModel">The owning DynamoModel object.</param>
+        /// 
+        internal PulseMaker(DynamoModel dynamoModel)
+        {
+            this.dynamoModel = dynamoModel;
+            this.dynamoModel.EvaluationCompleted += OnRunExpressionCompleted;
```

Comment:
We need to stop introducing cyclic dependencies `DynamoModel` like this.  Because of this dependency, we can't write `PulseMaker` unit tests!  That's really bad.